test/os.cc

```diff
@@ -0,0 +1,87 @@
+#include "eventuals/os.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace eventuals::test {
+namespace {
+
+#if defined(__linux__) || defined(__MACH__)
+
+using eventuals::os::Thread;
+using testing::MockFunction;
+
+void foo() {
+  // Do nothing.
+}
+
+static bool functor_was_called = false;
+
+struct Moveable {
+  Moveable() = default;
+  ~Moveable() = default;
+  Moveable(const Moveable& c) = delete;
+  Moveable& operator=(const Moveable&) = delete;
+  Moveable(Moveable&&) {}
+
+  void operator()() {
+    functor_was_called = true;
+  }
+};
+
+TEST(Thread, NotJoinable) {
+  Thread t;
+  EXPECT_FALSE(t.is_joinable());
+}
+
+TEST(Thread, Joinable) {
+  Thread t{&foo, "thread_name"};
+  EXPECT_TRUE(t.is_joinable());
+  t.join();
+  EXPECT_FALSE(t.is_joinable());
+}
+
+TEST(Thread, LambdaThatCapturesEverything) {
+  MockFunction<void()> start;
+
+  EXPECT_CALL(start, Call())
+      .Times(1);
+
+  Thread t{[&]() {
+             start.Call();
+           },
+           "thread_name",
+           4'194'304};
```

Comment:
```suggestion
           Megabytes(4)};
```

Reply:
Will use Bytes(4'194'304) instead. Again, on macOS stack size should be a multiple of the system page size!